numba/np/random/distributions.py

```diff
@@ -19,45 +19,28 @@
 from numba.np.random.generator_core import (next_double, next_float,
                                             next_uint32, next_uint64)
 from numba import float32, int64
-from numba.np.numpy_support import numpy_version
 # All of the following implementations are direct translations from:
 # https://github.com/numpy/numpy/blob/7cfef93c77599bd387ecc6a15d186c5a46024dac/numpy/random/src/distributions/distributions.c
 
 
-if numpy_version >= (1, 21):
-    @register_jitable
-    def np_log1p(x):
-        return np.log1p(x)
-
-    @register_jitable
-    def np_log1pf(x):
-        return np.log1p(float32(x))
-
-    @register_jitable
-    def random_rayleigh(bitgen, mode):
-        return mode * np.sqrt(2.0 * random_standard_exponential(bitgen))
-else:
-    @register_jitable
-    def np_log1p(x):
-        return np.log(1.0 + x)
```

Comment:
How did this ever match the precision of the NumPy `log1p`?

Reply:
(Sorry, that's a bit of a rhetorical question - an expression of surprise, rather than anything that needs addressing / discussion)